imap_processing/cdf/cdf_attribute_manager.py

```diff
@@ -0,0 +1,192 @@
+            const.CDF_EPOCH.value,
+            const.CDF_EPOCH16.value,
+            const.CDF_TIME_TT2000.value,
+        ]
+
+        # Load Default IMAP Global Attributes
+        self.global_attributes = self._load_yaml_data(
+            str(self.source_dir / DEFAULT_GLOBAL_CDF_ATTRS_FILE)
+        )
+        self.variable_attributes = dict()
+
+    def _load_default_global_attr_schema(self) -> dict:
+        default_schema_path = str(
+            self.source_dir
+            / "shared"
+            / DEFAULT_GLOBAL_CDF_ATTRS_SCHEMA_FILE
+        )
+        # Load the Schema
+        return self._load_yaml_data(default_schema_path)
+
+    def _load_default_variable_attr_schema(self) -> dict:
+        default_schema_path = str(
+            self.source_dir
+            / "shared"
+            / DEFAULT_VARIABLE_CDF_ATTRS_SCHEMA_FILE
+        )
+        # Load the Schema
+        return CdfAttributeManager._load_yaml_data(default_schema_path)
+
+    def load_global_attributes(self, file_path: str):
+        """
+        Update the global attributes property with the attributes from the file.
+
+        Parameters
+        ----------
+        file_path: str
+            file path to load, under self.source_dir.
+
+        """
+        self.global_attributes.update(
+            self._load_yaml_data(
+                str(self.source_dir / file_path)
+            )
+        )
+
+    @staticmethod
+    def _load_yaml_data(file_path: str | Path) -> dict:
+        """
+        Load a yaml file from the provided path.
+
+        Parameters
+        ----------
+        file_path: str | Path
+            path to the yaml file to load.
+
+        Returns
+        -------
+            Loaded yaml.
+        """
+        with open(file_path, "r") as file:
+            return yaml.safe_load(file)
+
+    def global_attribute_template(self, logical_source_id=None) -> OrderedDict:
+        """
+        Function to generate a template of required global attributes
+        that must be set for a valid CDF.
+
+        If a logical_source_id is provided, the level and instrument specific
+        attributes that were previously loaded using add_instrument_global_attrs will
+        be included.
+
+        Parameters
+        ----------
+        logical_source_id : str
+            The logical source id of the CDF file, used to retrieve instrument and level
+            specific global attributes.
+
+        Returns
+        -------
+        template : `OrderedDict`
+            A template for required global attributes that must be provided.
+        """
+        # TODO: Move this validation into the load step
+        template = OrderedDict()
+        for attr_name, attr_schema in self.global_attribute_schema.items():
+            if (
+                attr_schema["required"]
+                and not attr_schema["derived"]
+                and attr_name not in self.global_attributes
+            ):
+                # TODO throw an error?
```

Comment:
Seems like if it is required, yes we should error. Maybe add a warning for now?
```suggestion
                # TODO throw an error?
                warnings.warn(f"Missing {attr_name} from the template")
```

Reply:
Well, right now we're using a schema that might not match, so it will always throw a warning